### ISSUE TITLE
brace-style does not validate the closing curly brace in switch statements

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#7869]
**Tell us about your environment**

* **ESLint Version:** master
* **Node Version:** 7.3.0
* **npm Version:** 3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?**

**Please show your full configuration:**

(none)

**What did you do? Please include the actual source code causing the issue.**

```js
/* eslint brace-style: error */

if (foo) {
  bar() }

switch (foo) {
  case 1: bar() }
```

**What did you expect to happen?**

I expected two errors to be reported, one for each of the closing `}` braces.

**What actually happened? Please include the actual, raw output from ESLint.**

Only one error was reported:

```
4:3  error  Closing curly brace should be on the same line as opening curly brace or on the line after the previous block  brace-style
```

**Comment by not-an-aardvark** [comment]
Additionally, the `allowSingleLine` option does not seem to work for `SwitchStatement` nodes:

```js
/* eslint brace-style: [error, "allman", {"allowSingleLine": true}] */

switch (a) { }
```

Expected: No error

Actual result:

```
3:1 - Opening curly brace appears on the same line as controlling statement. (brace-style)
```
